### PR TITLE
Add memory limits to dev Docker stack containers

### PR DESCRIPTION
**Pull request by Scotchester**
Per Lamin's suggestion at https://GHE/CFGOV/platform/issues/3979

## Checklist

- [x] PR has an informative and human-readable title
  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)

### CODE REVIEW

File: docker-stack.yml [hunk 1]
```diff
@@ -29,8 +35,14 @@ services:
         delay: 10s
         max_attempts: 3
         window: 120s
+      resources:
```
Comment (alexm118) [inline docker-stack.yml:38]: Your elasticsearch service failed to start in the docker cluster - my guess is this is insufficient allocation for the container. [Guidelines](https://www.elastic.co/guide/en/elasticsearch/reference/current/docker.html) don't seem to specify a requirement, maybe we can look at usage stats for other stacks and see what they are.

Comment (Scotchester) [reply docker-stack.yml:38]: Trying 2 GB limits with 1 GB reservations. We'll see how the next deployment goes.

**Review by higs4281 — APPROVED** [review]
We could probably whack the Elasticsearch2 services way back, since they're unused in almost all our stacks. Can't wait to drop them entirely. But I think these limits are worth a try.

**Review by cwdavies — CHANGES_REQUESTED** [review]
There might be issues adding the --user option to the docker run command in the Jenkinsfile.
This seems out of scope in this PR to run a container as a specific user and should not impact memory limitations.

**Comment by Scotchester** [comment]
> There might be issues adding the --user option to the docker run command in the Jenkinsfile.
> This seems out of scope in this PR to run a container as a specific user and should not impact memory limitations.

Yeah, I'm gonna revert that. I thought it might have been the solution to this job's workspace being undeletable, but then I realized that it has to be set to begin with so that the created screenshots are created with that user.

**Review by cwdavies — APPROVED** [review]
These memory limits and reservations look good to me in docker-stack.yml

**Pull request merged**